### ISSUE TITLE
`v self` doesn't work if the binary is not "v"

### ISSUE THREAD

**Issue by medvednikov** [vlang/v#11069]
![image](https://user-images.githubusercontent.com/687996/128425421-67aff4bf-209b-4337-8231-721cdf56f03d.png)

**Comment by brightly-salty** [comment]
I think this bug is much wider than just `v self`. In what I tested, for example, I found that it also occurs in `v up` and `v test-all` and I suspect the bug is the result of consistent prioritization of `$VEXE` over the executable actually called.

**Comment by yuyi98** [comment]
Do we need to check the v compiler file name？

**Comment by JalonSolov** [comment]
Should just be able to use `arg[0]` instead of forcing `$VEXE`.

**Comment by brightly-salty** [comment]
In this case, we are re-calling `vself` using `utils.launch_tool` so `arg[0]` points to the location of `vself` not `v`. Maybe we could pass `arg[0]` using `utils.launch_tool` as a CL argument?

**Comment by JalonSolov** [comment]
Yes, that would be one way.  `vself` is almost never executed directly, is it usually always `v self`.  If someone does run `vself` directly... "You pays yer money and you takes yer chances."

**Comment by medvednikov** [comment]
Fixed.
